lib/cli/src/generators/baseGenerator.ts

```diff
@@ -64,21 +64,29 @@ export async function baseGenerator(
   const yarn2Dependencies =
     packageManager.type === 'yarn2' ? ['@storybook/addon-docs', '@mdx-js/react'] : [];
 
+  const packageJson = packageManager.retrievePackageJson();
+  const alreadyInstalledDependencies = [
+    ...Object.keys(packageJson.dependencies),
+    ...Object.keys(packageJson.devDependencies),
+  ];
+
   const packages = [
     `@storybook/${framework}`,
     ...addonPackages,
     ...extraPackages,
     ...extraAddons,
     ...yarn2Dependencies,
-  ].filter(Boolean);
+  ]
+    .filter(Boolean)
+    .filter((packageToInstall) => !alreadyInstalledDependencies.includes(packageToInstall));
```

Comment:
As of a recent PR from me, these packages might be something like `vue-loader@^16` instead of just the package name. Do we need to account for that?

Reply:
Interesting! I will update the code split it and keep only the pre-`@` part. Thanks!

Reply:
👍 There is logic in CLI to do this that you can pull into a utility: https://github.com/storybookjs/storybook/blob/next/lib/cli/src/js-package-manager/JsPackageManager.ts#L11-L21

Reply:
I decided to check only the regular deps because if a package is already installed as devDep it will be updated without any issue (the problem is when you try to install regular dep as dev dep and dev dep as regular dep)